banyan/src/zstd_dag_cbor_seq.rs

```diff
@@ -246,19 +248,25 @@ impl ZstdDagCborSeq {
     }
 
     /// encrypt using the given key and nonce
-    pub fn encrypt(&self, nonce: &salsa20::XNonce, key: &salsa20::Key) -> anyhow::Result<Vec<u8>> {
+    pub fn encrypt(
+        &self,
+        nonce: &chacha20::XNonce,
+        key: &chacha20::Key,
+    ) -> anyhow::Result<Vec<u8>> {
         self.clone().into_encrypted(nonce, key)
     }
 
     /// convert into an encrypted blob, using the given key and nonce
     pub fn into_encrypted(
         self,
-        nonce: &salsa20::XNonce,
-        key: &salsa20::Key,
+        nonce: &chacha20::XNonce,
+        key: &chacha20::Key,
     ) -> anyhow::Result<Vec<u8>> {
         let Self { mut data, links } = self;
         // encrypt in place with the key and nonce
-        XSalsa20::new(key, nonce).apply_keystream(&mut data);
+        let mut chacha20 = XChaCha20::new(key, nonce);
+        chacha20.seek(0u64);
+        chacha20.apply_keystream(&mut data);
```

Comment:
not sure we need the seek

Reply:
we don't. Was a leftover from the more ambitious stuff...